src/integration/java/com/salesforce/dynamodbv2/mt/admin/AmazonDynamoDbAdminUtilsIt.java

```diff
@@ -0,0 +1,122 @@
+import com.amazonaws.services.dynamodbv2.util.TableUtils;
+import com.salesforce.dynamodbv2.dynamodblocal.AmazonDynamoDbLocal;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class AmazonDynamoDbAdminUtilsIt {
+
+    /* This tests against hosted DynamoDB to ensure tables encounter TableInUse exception
+    (local create times are too short) */
+
+    static final Regions REGION = Regions.US_EAST_1;
+    AmazonDynamoDB remoteDynamoDB = AmazonDynamoDBClientBuilder.standard()
+            .withCredentials(new EnvironmentVariableCredentialsProvider())
+            .withRegion(REGION).build();
+
+    AmazonDynamoDbAdminUtils remoteUtils = new AmazonDynamoDbAdminUtils(remoteDynamoDB);
+
+    AmazonDynamoDB localDynamoDB = AmazonDynamoDbLocal.getAmazonDynamoDbLocal();
+    AmazonDynamoDbAdminUtils localUtils = new AmazonDynamoDbAdminUtils(localDynamoDB);
+    private static final String TABLE_PREFIX = "oktodelete-testBillingMode.";
+    String tableName;
+    String fullTableName;
+
+    @BeforeEach
+    void beforeEach() {
+        tableName = new String(String.valueOf(System.currentTimeMillis()));
+        fullTableName = TABLE_PREFIX + tableName;
+    }
+
+    @Test
+    void createTableIfNotExists() {
+    }
+
+    private CreateTableRequest getTestCreateTableRequest(String tableName) {
+
+        return new CreateTableRequest()
+                .withTableName(tableName)
+                .withAttributeDefinitions(new AttributeDefinition(HASH_KEY_FIELD, S),
+                        new AttributeDefinition(RANGE_KEY_FIELD, S),
+                        new AttributeDefinition(INDEX_FIELD, S))
+                .withKeySchema(new KeySchemaElement(HASH_KEY_FIELD, KeyType.HASH),
+                        new KeySchemaElement(RANGE_KEY_FIELD, KeyType.RANGE))
+                .withProvisionedThroughput(new ProvisionedThroughput(1L, 1L))
+                .withGlobalSecondaryIndexes(new GlobalSecondaryIndex().withIndexName("testgsi")
+                        .withKeySchema(new KeySchemaElement(INDEX_FIELD, KeyType.HASH))
+                        .withProvisionedThroughput(new ProvisionedThroughput(1L, 1L))
+                        .withProjection(new Projection().withProjectionType(ProjectionType.ALL)))
+                .withLocalSecondaryIndexes(new LocalSecondaryIndex().withIndexName("testlsi")
+                        .withKeySchema(new KeySchemaElement(HASH_KEY_FIELD, KeyType.HASH),
+                                new KeySchemaElement(INDEX_FIELD, KeyType.RANGE))
+                        .withProjection(new Projection().withProjectionType(ProjectionType.ALL)));
+    }
+
+    @Test
+    void createTableIfNotExistsButTableInUseForCreating() {
+
+        remoteDynamoDB.createTable(getTestCreateTableRequest(fullTableName));
+        remoteUtils.createTableIfNotExists(getTestCreateTableRequest(fullTableName), 10);
+
+        // cleanup
+        remoteUtils.deleteTableIfExists(fullTableName, 10, 600);
+    }
```

Comment:
Should the clean up be in @After or @AfterClass?

Reply:
moved all cleanup to afterAll